dotnet-gitchanges/Readers/Parsers/DefaultCommitParser.cs

```diff
@@ -9,24 +10,39 @@ namespace Gitchanges.Readers.Parsers
     public class DefaultCommitParser : ICommitParser<IChange>
     {
         private const string Unreleased = "Unreleased";
+        private const string Uncategorized = "Uncategorized";
         private readonly ParsingPatterns _patterns;
         private string _lastVersion = Unreleased;
+        private IDictionary<string, string> _commitShaToTagName;
         
-        public DefaultCommitParser(ParsingPatterns patterns)
+        public DefaultCommitParser(ParsingPatterns patterns, IDictionary<string, string> commitShaToTagName = null)
         {
+            _commitShaToTagName = commitShaToTagName;
             _patterns = patterns;
         }
         
         public IChange Parse(Commit commit)
         {
             var reference = GetMatchOrDefault(Regex.Match(commit.Message, _patterns.Reference));
-            var version = GetMatchOrDefault(Regex.Match(commit.Message, _patterns.Version));
-            var tag = GetMatchOrDefault(Regex.Match(commit.Message, _patterns.Tag));
-                    
-            if (string.IsNullOrEmpty(version) || string.IsNullOrEmpty(tag)) return null;
+            var tag = GetMatchOrDefault(Regex.Match(commit.Message, _patterns.Tag), _commitShaToTagName != null ? Uncategorized : "");

```

Comment:
I'd like to see a test covering the "default" case. (i.e. when there is no tag match and the dictionary is not null)

Reply:
Added